src/tests/ftest/cart/util/daos_sys_logscan.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+#
+# (C) Copyright 2024 Intel Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+
+"""Scan daos_engine log files to get a summary of pools activity."""
+
+import argparse
+import re
+import sys
+import time
+import pprint
+from collections import Counter, OrderedDict, defaultdict
+
+import cart_logparse
+
+class SysPools():
+    """Directory of Pools and Summary Activities Found in Engine Log Files"""
+
+    # TODO
+    # diagram of nested dictionaries constructed
+    # system map update events (output outside if pool-specific context)
+    # SWIM events seen by PS leader?
+    # add/remove target events on PS leader?
+    # pool map version changes
+    # rebuild queued (PS leader)
+    # rebuild scanning (PS leader, and count of # engines completing scanning, time duration, stalls)
+    # rebuild pulling (PS leader? count of # of engines starting/finishing pulling, may be a subset, time duration, stalls)
+    # rebuild number of objects, records and progress made?
+    # rebuild aborted/errored
+    # rebuild completed/success
+
+    re_rank_assign = re.compile("ds_mgmt_drpc_set_rank.*set rank to (\d+)")
```

Comment:
`W605 invalid escape sequence` usually resolved by using a regex string
```suggestion
    re_rank_assign = re.compile(r"ds_mgmt_drpc_set_rank.*set rank to (\d+)")
```